include/gpio.h

```diff
@@ -189,6 +189,20 @@ extern "C" {
  */
 #define GPIO_DS_DISCONNECT_HIGH (0x3 << GPIO_DS_HIGH_POS)
 
+/** @cond INTERNAL_HIDDEN */
+#define GPIO_DMA_REQ_POS		15
+#define GPIO_DMA_REQ_MASK		(0x3 << GPIO_DMA_REQ_POS)
+/** @endcond */
+
+/** GPIO pin trigger DMA requenst on falling edge. */
+#define GPIO_DMA_REQ_EDGE_LOW		(1 << GPIO_DMA_REQ_POS)
+
+/** GPIO pin trigger DMA requenst on rising edge. */
+#define GPIO_DMA_REQ_EDGE_HIGH		(2 << GPIO_DMA_REQ_POS)
+
+/** GPIO pin trigger DMA requenst on rising and falling edge. */
+#define GPIO_DMA_REQ_EDGE_DOUBLE	(3 << GPIO_DMA_REQ_POS)
+
```

Comment:
Are those generic enough to be in the API?

Reply:
It's also possible on LPC MCUs, so it does make sense to include in the API.